### ISSUE TITLE
Comparison operation to null

### ISSUE THREAD

**Issue by paolosanchi** [crcn/sift.js#255]
I found an unexpected behaviour the comparison operation and null fields.

`sift({ name: { "$eq": null } })({ name: null });`

this correclty returns true, since null is equal to null

`sift({ name: { "$lte": null } })({ name: null })`

This should also return true, because, but it doesn't.

The culprit seems to me the following line
https://github.com/crcn/sift.js/blob/e1c4e1a0f0b681b2f48db6e963906ac630acb7ed/src/core.ts#L308

`sift({ name: { "$gte": null } })({ name: "text" })`

This case also returns false

This may be correct, because javascript itself does the same, but I think it may be arguably true, CouchDb and it's MangoDb query language works like that, the sorting (and so comparison) it is based on this collation specification https://db-staging.resolvi.cloud/_utils/docs/ddocs/views/collation.html#collation-specification

**Comment by crcn** [comment]
Finally getting around to this. Fix should be in and published to NPM as `v17.0.1`. 

Thanks for catching this, let me know if there's anything else!

**Comment by paolosanchi** [comment]
Hi!
thank you for fixing.
When I faced this I did an override to the operators and used the collate comparison function from pouchdb-collate, and ported the same operators from pouchdb-find, so I am sure to have the very same results.

Hope this may be useful:

```
import sift, { createEqualsOperation } from "sift";
import { collate } from 'pouchdb-collate';

// https://github.com/pouchdb/pouchdb/blob/0bdb3423acf0cc4fae8747c1bd2cea0b1cf23339/packages/node_modules/pouchdb-selector-core/src/in-memory-filter.js#L254

//  '$eq': function (doc, userValue, parsedField, docFieldValue) {
//     return fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) === 0;
//   },

//   '$gte': function (doc, userValue, parsedField, docFieldValue) {
//     return fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) >= 0;
//   },

//   '$gt': function (doc, userValue, parsedField, docFieldValue) {
//     return fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) > 0;
//   },

//   '$lte': function (doc, userValue, parsedField, docFieldValue) {
//     return fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) <= 0;
//   },

//   '$lt': function (doc, userValue, parsedField, docFieldValue) {
//     return fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) < 0;
//   },
export function createCollatedSift(selector) {
    return sift(selector, {
        operations: {
            $eq(userValue, ownerQuery, options) {
                return createEqualsOperation(
                    docFieldValue => fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) === 0,
                    ownerQuery,
                    options
                );

            },
            $gte(userValue, ownerQuery, options) {
                return createEqualsOperation(
                    docFieldValue => fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) >= 0,
                    ownerQuery,
                    options
                );
            },
            $gt(userValue, ownerQuery, options) {
                return createEqualsOperation(
                    docFieldValue => fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) > 0,
                    ownerQuery,
                    options
                );
            },
            $lte(userValue, ownerQuery, options) {
                return createEqualsOperation(
                    docFieldValue => fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) <= 0,
                    ownerQuery,
                    options
                );
            },
            $lt(userValue, ownerQuery, options) {
                return createEqualsOperation(
                    docFieldValue => fieldIsNotUndefined(docFieldValue) && collate(docFieldValue, userValue) < 0,
                    ownerQuery,
                    options
                );
            },
        }
    });
}


function fieldIsNotUndefined(docFieldValue) {
    return typeof docFieldValue !== 'undefined';
}
```
